### PR TITLE
feat: added code for feedback feature

### PR DESCRIPTION
**Pull request by Bhargav-Dave**

### CODE REVIEW

**Comment by dennyabrain** [comment]
@Bhargav-Dave : the code in `sendFeedback()` looks ok to me. Can you tell me what issue you are facing with making the api call ?

**Comment by Bhargav-Dave** [comment]
> @Bhargav-Dave : the code in `sendFeedback()` looks ok to me. Can you tell me what issue you are facing with making the api call ?

@dennyabrain I think the problem comes in defining UserData, I was reviewing the logs and caught the error now. If you notice the function uses UserData but it is not defined in the function itself. Let me try again with the correct definitions

**Comment by Bhargav-Dave** [comment]
Meanwhile can you please check the code in api-server/index.js and let me know if that is fine? @dennyabrain

**Comment by Bhargav-Dave** [comment]
> > @Bhargav-Dave : the code in `sendFeedback()` looks ok to me. Can you tell me what issue you are facing with making the api call ?
> 
> @dennyabrain I think the problem comes in defining UserData, I was reviewing the logs and caught the error now. If you notice the function uses UserData but it is not defined in the function itself. Let me try again with the correct definitions

Doing this fixes the error but I cannot make the API call to the backend, but running into error 501, the problem is probably in the index.js code

File: browser-extension/api-server/index.js [hunk 1]
```diff
@@ -56,6 +58,32 @@ app.post("/preference/", async (req, res) => {
...
+      userId: req.user_id,
+      tweetText: req.tweet_text,
+      sentiment: req.tweet_sentiment,
+      confidence: req.tweet_confidence
+      
+    });
+
+    res.send({ msg: "Feedback Sent" });
+  } catch (err) {
+    res.status(501).send({ msg: "Error sending feedback" });
```
Comment (dennyabrain) [inline browser-extension/api-server/index.js:80]: @Bhargav-Dave 
check the output of console.log(err) before this line. that will tell you the exact error

Comment (Bhargav-Dave) [reply browser-extension/api-server/index.js:80]: The error got resolved.
The api call is working fine and the data is being sent!
